components/Header/index.js

```diff
@@ -0,0 +1,93 @@
+import { useTransactionsContext } from "@/utils/TransactionsContext/TransactionsContext";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+
+export default function Header() {
+    const router = useRouter();
+    const activeNavPoint = router.pathname;
+    const { handleFilterChange } = useTransactionsContext();
```

Comment:
It is not a good practice to let the header handle any data. Please try to seperate concerns in a more structured way. The headers purpose should only be to render a header that includes a link to the homepage. Any other concerns should be handled somewhere else, e.g.
- The navigation has to handle the highlighting of the actual page
- The pages/index.js has to handle its default state